data/data.go

```diff
@@ -2,68 +2,56 @@ package data
 
 import (
 	"context"
-	"crypto/rand"
-	dbsql "database/sql"
+	"database/sql"
 	"fmt"
-	"log"
 	"time"
 
-	"github.com/facebookincubator/ent/dialect/sql"
+	entsql "github.com/facebookincubator/ent/dialect/sql"
 	"github.com/pdeguing/empire-and-foundation/ent"
+	"github.com/pdeguing/empire-and-foundation/ent/migrate"
 	"github.com/pkg/errors"
-
-	_ "github.com/lib/pq"
 )
 
+// Client is and ent ORM client with which database queries can be made.
 var Client *ent.Client
-var DB *dbsql.DB
 
-func open() (*ent.Client, error) {
-	drv, err := sql.Open("postgres", "dbname=empire_and_foundation sslmode=disable")
+// DB is the raw SQL DB handle to the database. It is recommended to use
+// Client, but in some cases the raw handle is needed. For instance, in
+// use of external libraries.
+var DB *sql.DB
+
+// InitDatabaseConnection makes a connection to the database and
+// sets Client and DB so they can be used by the rest of the application.
+func InitDatabaseConnection(driver, source string, debug bool) error {
+	drv, err := entsql.Open(driver, source)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed opening database connection: %w", err)
 	}
+
 	// Get the underlying sql.DB object of the driver.
 	DB = drv.DB()
 	DB.SetMaxIdleConns(10)
 	DB.SetMaxOpenConns(100)
 	DB.SetConnMaxLifetime(time.Hour)
-	return ent.NewClient(ent.Driver(drv)), nil
-}
 
-func init() {
-	var err error
-	Client, err = open()
-	if err != nil {
-		log.Fatalf("failed opening connection to postgres: %v", err)
-	}
-	err = Migrate(context.Background(), Client)
-	if err != nil {
-		log.Fatalf("failed creating schema resources: %v", err)
+	Client = ent.NewClient(ent.Driver(drv))
+	if debug {
+		Client = Client.Debug()
 	}
-	return
+	return nil
 }
 
 // Migrate runs the database schema migrations.
-func Migrate(ctx context.Context, client *ent.Client) error {
-	return client.Schema.Create(ctx)
-}
-
-// create a random UUID from RFC 4122
-func createUUID() (uuid string) {
-	u := new([16]byte)
-	_, err := rand.Read(u[:])
+func Migrate(ctx context.Context, client *ent.Client, dropIndex, dropColumn bool) error {
+	err := client.Schema.Create(
+		ctx,
+		migrate.WithDropIndex(dropIndex),
```

Comment:
Weird I thought I added it before

Reply:
You did. But you removed it or didn't commit it.